mypyc/test-data/run-valuetype.test

```diff
@@ -0,0 +1,336 @@
+[case testValueTypeBasic]
+from typing import final, Final
+from mypy_extensions import mypyc_attr, i32
+
+@final
+@mypyc_attr(value_type=True)
+class Vector2I:
+    def __init__(self, x: i32, y: i32) -> None:
+        self.x: Final = x
+        self.y: Final = y
+
+    def __add__(self, other: "Vector2I") -> "Vector2I":
+        return Vector2I(self.x + other.x, self.y + other.y)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Vector2I):
+            return NotImplemented
+        return self.x == other.x and self.y == other.y
+
+def test_basic_value_type() -> None:
+    a = Vector2I(1, 2)
+    b = Vector2I(3, 4)
+    c = a.__add__(b)
+    assert c == Vector2I(4, 6)
+
+def test_value_type_operator_is() -> None:
+    # The 'is' operator have undefined result for value types.
+    # however it must not raise an error.
+    Vector2I(1, 2) is Vector2I(1, 2)
+
+[case testValueTypeInheritance]
+from typing import final, Final
+from mypy_extensions import mypyc_attr, i32
+
+class A:
+    def __init__(self, x: i32) -> None:
+        self.x: Final = x
+
+    def method(self) -> i32:
+        return self.x + 4
+
+    def method2(self) -> i32:
+        return self.x + 4
+
+    @property
+    def prop(self) -> i32:
+        return self.x + 4
+
+    @property
+    def prop2(self) -> i32:
+        return self.x + 4
+
+@final
+@mypyc_attr(value_type=True)
+class B(A):
```

Comment:
I'm not sure if it makes sense for value types to subclass non-value types (other than the implicit `object`). This could break LSP in some use cases that assume that object identity is preserved across the lifetime of an object. For example, assume that some extra information is stored in some dict based on object identity. Now if using a value type subclass, that extra information can be lost when the value is passed around. 

Also, calling inherited methods seems tricky, and currently seems to generate compile errors:
```py
from typing import final, Final
from mypy_extensions import mypyc_attr

class Bar:
    def __init__(self) -> None:
        self.xx: Final = 0

    def method(self) -> int:
        return self.xx


@final
@mypyc_attr(value_type=True)
class Foo(Bar):
    def __init__(self) -> None:
        self.yy: Final = 0


def foo(f: Foo) -> int:
    return f.method()  # This seems to generate bad C (didn't investigate in detail)
```

Reply:
You are right, I will change this to support only inheritance on other value types only.